src/main/java/uk/ac/cam/cl/dtg/segue/scheduler/jobs/DeleteEventAdditionalBookingInformationJob.java

```diff
@@ -0,0 +1,120 @@
+        // Magic number
+        Integer limit = 10000;
+        Integer startIndex = 0;
+        Map<String, List<String>> fieldsToMatch = Maps.newHashMap();
+        Map<String, Constants.SortOrder> sortInstructions = Maps.newHashMap();
+        Map<String, AbstractFilterInstruction> filterInstructions = Maps.newHashMap();
+        fieldsToMatch.put(TYPE_FIELDNAME, Collections.singletonList(EVENT_TYPE));
+        sortInstructions.put(DATE_FIELDNAME, SortOrder.DESC);
+        DateRangeFilterInstruction anyEventsToNow = new DateRangeFilterInstruction(null, new Date());
+        filterInstructions.put(ENDDATE_FIELDNAME, anyEventsToNow);
+        ZonedDateTime now = ZonedDateTime.now();
+        ZonedDateTime thirtyDaysAgo = now.plusDays(-30);
+        try {
+            ResultsWrapper<ContentDTO> findByFieldNames = this.contentManager.findByFieldNames(
+                    properties.getProperty(CONTENT_INDEX), ContentService.generateDefaultFieldToMatch(fieldsToMatch),
+                    startIndex, limit, sortInstructions, filterInstructions);
+            for (ContentDTO contentResult : findByFieldNames.getResults()) {
+                if (contentResult instanceof IsaacEventPageDTO) {
+                    IsaacEventPageDTO page = (IsaacEventPageDTO) contentResult;
+                    // Event end date (if present) > 30 days ago, else event date > 30 days ago
+                    if ((page.getEndDate() != null && page.getEndDate().toInstant().isBefore(thirtyDaysAgo.toInstant())) || (page.getDate().toInstant().isBefore(thirtyDaysAgo.toInstant()))) {
+                        try (Connection conn = database.getDatabaseConnection()) {
+                            PreparedStatement pst;
+                            // Check for additional info that needs removing, check if pii has already been removed, if
+                            // so, don't re-remove
+                            pst = conn.prepareStatement(
+                                    "UPDATE event_bookings SET additional_booking_information=jsonb_set(jsonb_set(jsonb_set(jsonb_set(additional_booking_information," +
+                                    " '{emergencyName}', '\"[REMOVED]\"'::JSONB, FALSE)," +
+                                    " '{emergencyNumber}', '\"[REMOVED]\"'::JSONB, FALSE)," +
+                                    " '{accessibilityRequirements}', '\"[REMOVED]\"'::JSONB, FALSE)," +
+                                    " '{medicalRequirements}', '\"[REMOVED]\"'::JSONB, FALSE)," +
+                                    " pii_removed=? " +
+                                    " WHERE event_id = ?" +
+                                    " AND additional_booking_information ??| array['emergencyName', 'emergencyNumber', 'accessibilityRequirements', 'medicalRequirements']" +
+                                    " AND pii_removed IS NULL");
+                            pst.setTimestamp(1, new Timestamp(new Date().getTime()));
```

Comment:
In this class we use a mix of the old style java.util.Date and new style Instant stuff from java.time which makes me uneasy but it is also a problem throughout the code (we also use the Joda-Time library which is basically the same as the java.time module I think). I will add it to the list of tech debt because time and date is complicated enough as it is without mixing with semi-deprecated classes!